lib/task_bunny/message.ex

```diff
@@ -1,9 +1,42 @@
 defmodule TaskBunny.Message do
   @moduledoc """
-  Provides functionalities to access a message and its meta data.
+  Functions to access messages and its meta data.
   """
 
   @doc """
+  Encode message body in JSON with job and arugment.
+  """
+  @spec encode(atom, any) :: String.t
+  def encode(job, payload) do
+    %{
+      "job" => Atom.to_string(job),
+      "payload" => payload,
+      "created_at" => DateTime.utc_now()
+    }
+    |> Poison.encode!
+  end
+
+  @doc """
+  Decode message body in JSON to map
+  """
+  @spec decode(String.t) :: map
+  def decode(message) do
+    case Poison.decode(message) do
+      {:ok, decoded} ->
+        job = String.to_atom(decoded["job"])
```

Comment:
I might not be understanding your point fully but this test is passing: https://github.com/shinyscorpion/task_bunny/blob/a94412c435fb62d482fee0a154c355abf80df9f0/test/task_bunny/message_test.exs#L12-L16

Good point on memory leak and suggestion on using `to_existing_atom` 👍